### PR TITLE
[s2s]: script to convert pl checkpoints to hf checkpoints

### PR DESCRIPTION
**Pull request by sshleifer**

### CODE REVIEW

File: examples/seq2seq/convert_pl_checkpoint_to_hf.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        new_sd[k] = new_t
+    return new_sd
+
+
+def convert_pl_to_hf(pl_ckpt_path: str, hf_src_model_dir: str, save_path: str) -> None:
+    """Cleanup a pytorch-lightning .ckpt file or experiment dir and save a huggingface model with that state dict.
+    Silently allows extra pl keys (like teacher.) Puts all ckpt models into CPU RAM at once!
+
+    Args:
+        pl_ckpt_path: (str) path to a .ckpt file saved by pytorch_lightning or dir containing ckpt files.
```
Comment (sgugger) [inline examples/seq2seq/convert_pl_checkpoint_to_hf.py:40]:
```suggestion
        pl_ckpt_path (:obj:`str`): Path to a .ckpt file saved by pytorch_lightning or dir containing ckpt files.
```

File: examples/seq2seq/convert_pl_checkpoint_to_hf.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            If a directory is passed, all .ckpt files inside it will be averaged!
+        hf_src_model_dir: (str) path to a directory containing a correctly shaped checkpoint
```
Comment (sgugger) [inline examples/seq2seq/convert_pl_checkpoint_to_hf.py:42]:
```suggestion
        hf_src_model_dir (:obj:`str`): Path to a directory containing a correctly shaped checkpoint
```

File: examples/seq2seq/convert_pl_checkpoint_to_hf.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        save_path: (str) directory to save the new model
```
Comment (sgugger) [inline examples/seq2seq/convert_pl_checkpoint_to_hf.py:43]:
```suggestion
        save_path (:obj:`str`): Directory to save the new model
```

File: examples/seq2seq/convert_pl_checkpoint_to_hf.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    hf_model = AutoModelForSeq2SeqLM.from_pretrained(hf_src_model_dir)
+    if os.path.isfile(pl_ckpt_path):
+        ckpt_files = [pl_ckpt_path]
+    else:
+        assert os.path.isdir(pl_ckpt_path)
+        ckpt_files = list(Path(pl_ckpt_path).glob("*.ckpt"))
+        assert ckpt_files, f"could not find any ckpt files inside the {pl_ckpt_path} directory"
+
+    if len(ckpt_files) > 1:
+        logger.info(f"averaging {ckpt_files}")
```
Comment (sgugger) [inline examples/seq2seq/convert_pl_checkpoint_to_hf.py:55]:
```suggestion
        logger.info(f"averaging the weights of {ckpt_files}")
```
(Without precision, I understand averaging predictions.)

**Review by sgugger — APPROVED** [review]
LGTM, just have some nits.

**Review by LysandreJik — APPROVED** [review]
LGTM

**Pull request merged**